go/vt/vtgate/vindexes/hash.go

```diff
@@ -69,10 +69,23 @@ func (vind *Hash) IsFunctional() bool {
 func (vind *Hash) Map(cursor VCursor, ids []sqltypes.Value) ([]key.Destination, error) {
 	out := make([]key.Destination, len(ids))
 	for i, id := range ids {
-		num, err := sqltypes.ToUint64(id)
-		if err != nil {
-			out[i] = key.DestinationNone{}
-			continue
+		var num uint64
+		var err error
+		if id.IsSigned() {
```

Comment:
This is better than what we have, but this will fail implicit conversions from string values. You could instead recreate (or reuse) this logic: https://github.com/vitessio/vitess/blob/master/go/sqltypes/arithmetic.go#L294, which will make this function more permissive.

I personally prefer more strict typing, but others have argued that we should more closely mimic MySQL. Let me know which way you want to go.

Reply:
That's how I originally did it, but apparently got confused along the way. See if it looks better now.